### PR TITLE
VirtualInbound: dont modify filter chains in place

### PR DESCRIPTION
**Pull request by rshriram**
Fixes https://github.com/istio/istio/issues/16393 where multiple
duplicate filters were introduced into the same filter chain. This
is caused by the fact that both the virtualInbound and the inboundListener
share the same filterChain object with same HCM. In addition, the 
existing virtual Inbound code was modifying the inbound filter chains
in place(!), resulting in both the inbound listener's filter chain and
the virtualInbound's filter chain having the same match values, when 
it should have been present only in the virtualInbound's filter chain.

Signed-off-by: Shriram Rajagopalan <rshriram@tetrate.io>

### CODE REVIEW

File: tests/integration/pilot/envoyfilter/main_test.go [hunk 1]
```diff
@@ -75,6 +74,46 @@ spec:
...
+      value:
+        typed_config:
+          "@type": "type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager"
+          access_log:
+          - name: envoy.http_grpc_access_log
+            config: 
+              common_config:
+                log_name: "grpc-als-example"
+                grpc_service:
+                  envoy_grpc:
```
Comment (brectanus-sigsci) [inline tests/integration/pilot/envoyfilter/main_test.go]: It seems this does not work with the envoy_grpc client as it is required to use a static cluster for ALS (works for ext_authz). See: https://github.com/istio/istio/issues/16393#issuecomment-523609369 for a working example.

Comment (rshriram) [reply tests/integration/pilot/envoyfilter/main_test.go]: Yes I am aware. But this is just a test, to ensure that changes deep in a HCM can be made.

**Comment by hzxuzhonghu** [comment]
should be careful when changing the input pointer type param

**Pull request merged**